R/tm_g_swimlane.R

```diff
@@ -309,26 +305,59 @@ srv_g_swimlane <- function(id,
 
     # create plot
     output_q <- reactive({
-      validate(need(iv$is_valid(), "Misspecification error: please observe red flags in the encodings."))
-
-      # DATA GETTERS
       validate(need("ADSL" %in% names(data), "'ADSL' not included in data"))
       validate(need(
         (length(data) == 1 && dataname == "ADSL") ||
-          (length(data) >= 2 && dataname != "ADSL"),
-        "Please either add just 'ADSL' as dataname when just ADSL is available
-      In case 2 datasets are available ADSL is not supposed to be the dataname."
+          (length(data) >= 2 && dataname != "ADSL"), paste(
+          "Please either add just 'ADSL' as dataname when just ADSL is available.",
+          "In case 2 datasets are available ADSL is not supposed to be the dataname."
+        )
       ))
 
       ADSL <- data[["ADSL"]]() # nolint
 
-      q1 <- teal.code::new_qenv(tdata2env(data), code = get_code_tdata(data))
+      anl_vars <- unique(c(
+        "USUBJID", "STUDYID",
+        input$marker_pos_var, input$marker_shape_var, input$marker_color_var
+      )) # nolint
+      adsl_vars <- unique(c(
+        "USUBJID", "STUDYID",
+        input$bar_var, input$bar_color_var, input$sort_var, input$anno_txt_var
+      ))
+
+      if (dataname == "ADSL") {
+        teal::validate_has_data(ADSL, min_nrow = 3)
+        teal::validate_has_variable(ADSL, adsl_vars)
+      } else {
+        anl <- data[[dataname]]()
+        teal::validate_has_data(anl, min_nrow = 3)
+        teal::validate_has_variable(anl, anl_vars)
+
+        validate(need(
+          !any(c(marker_pos_var, marker_shape_var, marker_color_var) %in% adsl_vars),
+          "marker-related variables need to come from marker data"
+        ))
+      }
+
+      iv <- shinyvalidate::InputValidator$new()
+      iv$add_rule("bar_var", shinyvalidate::sv_required(
+        message = "Bar Length is required"
+      ))
+      # If reference lines are requested
+      iv$add_rule("vref_line", ~ if (anyNA(as_numeric_from_comma_sep_str(.))) {
```

Comment:
I'm seeing lots of warnings in the console here:
![image](https://user-images.githubusercontent.com/15201933/206423808-6c1e18d4-0a5a-4ede-8459-4093c678b82a.png)

Reply:
Probably the case in waterfall as well?

Reply:
These are raised by `as.numeric` in `as_numeric_from_comma_sep_str`.